db/db_impl_compaction_flush.cc

```diff
@@ -977,21 +977,21 @@ void DBImpl::MaybeScheduleFlushOrCompaction() {
     return;
   }
 
-  while (unscheduled_flushes_ > 0 &&
-         bg_flush_scheduled_ < immutable_db_options_.max_background_flushes) {
+  auto bg_job_limits = GetBGJobLimits();
+  while (immutable_db_options_.max_background_flushes != 0 &&
```

Comment:
try to ignore the `immutable_db_options_.max_background_flushes != 0` for now. #2204 will fix it up (i just can't stack diffs).